src/cloud_provider/kubernetes.rs

```diff
@@ -704,6 +732,38 @@ pub fn check_kubernetes_has_enough_resources_to_deploy_environment(
     Ok(())
 }
 
+/// check that there is enough nodes before any charts deployment.
+pub fn check_kubernetes_has_enough_nodes_to_deploy<P>(
+    kubernetes_config: P,
+    envs: Vec<(&str, &str)>,
+    max_nodes: usize,
+) -> Result<(), SimpleError>
+where
+    P: AsRef<Path>,
+{
+    return match kubectl_exec_get_node(kubernetes_config, envs) {
+        Err(e) => {
+            error!("Can't get nodes before services deployments");
+            Err(e)
+        }
+        Ok(nodes) => match nodes.items.len() < (max_nodes * 80 / 100) {
```

Comment:
if you are already at max node, let's say 3;3. you will never be able to deploy something no ?

Reply:
yes that's the point, if more than 80% of nodes capacity is used, it's stuck

Reply:
ok for infra, but for env you will never be able to deploy something has you will have 3 nodes, but they will be empty

Reply:
If you're 3/3 yes, but you can be sure that at least 50% of the time you won't be able to deploy. So, the user will just have to upgrade the max capacity. That won't mean the cluster capacity will grow, but we reduce issues due to this point.